test/functional/testlib.bash

```diff
@@ -162,6 +237,22 @@ set_env_variables() {
 	validate_path "$env_name"
 	path=$(dirname "$(realpath "$env_name")")
 
+	export CLIENT_CERT_DIR="$TEST_NAME/target-dir/etc/swupd"
+	export CLIENT_CERT="$CLIENT_CERT_DIR/client.pem"
+	export CACERT_DIR="/tmp/swupd_test_certificates" # trusted key store path
+	export PORT_FILE="/tmp/$TEST_NAME-port_file.txt" # stores web server port
+	export SERVER_PID_FILE="/tmp/$TEST_NAME-pid_file.txt" # stores web server pid
+
+	# Add environment variables for PORT and SERVER_PID when web server used
+	if [ -f "$PORT_FILE" ]; then
+		export PORT=$(cat "$PORT_FILE")
```

Comment:
declaring a variable and assigning it a value in the same step seems to be considered a bad practice in bash: https://github.com/koalaman/shellcheck/wiki/SC2155